provider/core/src/hello_world.rs

```diff
@@ -184,23 +183,35 @@ impl IterableResourceProvider<HelloWorldV1Marker> for HelloWorldProvider {
     }
 }
 
-/// Adds entries to a [`HelloWorldProvider`] from [`AnyMarker`](crate::any::AnyMarker)
-impl crate::export::DataExporter<crate::any::AnyMarker> for HelloWorldProvider {
```

Comment:
Issue: I do not agree that this impl is "weird".  It is a useful example of how data can be exported into a memory store as opposed to serialized into a blob or filesystem.  Please restore the impl.

Reply:
It's not compatible with the changes I'm introducing here though. To make `DataExporter` `Sync` it needs to use internal mutability, as `put_payload` becomes non-mut. Do you want me to introduce a `RwLock` in `HelloWorldProvider` just so that this continues working?

Reply:
Actually we don't even have locks in `no_std`